src/Client.ts

```diff
@@ -280,12 +323,22 @@ export default class Client {
     this.knownPublicKeyBundles.delete(peerAddress)
   }
 
+  public async canMessage(peerAddress: string): Promise<boolean>
```

Comment:
Good call on the overloads here. I noticed there aren't overloads on the static method. We'll want those so that `Client.canMessage('foo')` has a return type of `boolean` instead of `boolean | boolean[]`